src/scripts/logic.js

```diff
@@ -0,0 +1,157 @@
+    localStorage.setItem(page, `${data}`);
+}
+
+const getDataFromLocalStorage = (page) => {
+
+    const pageData = localStorage.getItem(page);
+    return pageData;
+}
+
+const getSearchData = async (name) => {
+    let data = [];
+    let response;
+    if (getDataFromLocalStorage(name)) {
+
+        !getDataFromLocalStorage(name) && toggleEmptySearchResult(name);
+        appendTempalte(1, JSON.parse(getDataFromLocalStorage(name)));
+
+
+    } else {
+        data = await fetch(`https://swapi.dev/api/people/?search=${name}`);
+        response = await data.json();
+        !response.count && toggleEmptySearchResult(name);
+
+        setDataToLocalStarage(name, JSON.stringify(response.results));
+        appendTempalte(1, [...response.results])
+    }
+    toggleBackButton();
+}
+
+export const setPaginationTemplate = async () => {
+    const paginationContainer = document.getElementById("pagination");
+    const countOfPeople = await fetch(`https://swapi.dev/api/people/`);
+    const { count } = await countOfPeople.json();
+    const countOFPage = Math.ceil(count / 10);
+    paginationContainer.innerHTML = generatePaginationPages(countOFPage);
+    setEventsOnPaginationPages();
+}
+
+const generatePaginationPages = (countPages) => {
+    const pages = [];
+    for (let i = 1; i <= countPages; i++) {
+        pages.push(`<div ${i === 1 && "class='pagination-page'"} >${i}</div>`);
+    }
+    return pages;
+}
+
+const setEventsOnPaginationPages = () => {
+    const pages = Array.from(document.querySelectorAll(".pagination div"));
+
+    pages.forEach(page => {
+        page.addEventListener("click", function () {
+            document.getElementsByClassName("pagination-page")[0] && document.getElementsByClassName("pagination-page")[0].removeAttribute("class");
+            this.setAttribute("class", "pagination-page");
+            appendTempalte(this.innerText);
+        })
+    })
+
+
+}
+
+export const appendTempalte = async (page, searchData = null) => {
+    const container = document.getElementById("card-section");
+    const data = !searchData ? getDataFromLocalStorage(page) ? await JSON.parse(getDataFromLocalStorage(page)) : await getDataFromApi("people", page) : searchData;
+    const elements = [];
+    data.forEach((card) => {
+        elements.push(getCardTemplate(card));
+    });
+    container.innerHTML = elements;
+
+
+    Array.from(document.getElementsByClassName("primary-card"));
+    const cardsButton = Array.from(document.getElementsByClassName("primary-card__info-label"));
+    const closeElement = Array.from(
+        document.getElementsByClassName("fa-times-circle")
+    );
+    cardsButton.forEach((element) => {
+        element.addEventListener("click", setCardEvents);
+    });
+    closeElement.forEach((close) => {
+        close.addEventListener("click", function () {
+            this.closest(".primary-card").removeAttribute("id");
+        });
+    });
+
+
+}
+
+function setCardEvents() {
```

Comment:
Can you make it in this way?
`const setCardEvents = () =>{}`

Reply:
i need to use parent context!
this fucntion is using in addEventListener